Remotion/Core/Extensions/Logging/Log4Net/Log4NetLogger.cs

```diff
@@ -15,27 +15,118 @@
-    throw new NotImplementedException();
+    return logLevel switch
+    {
+        MicrosoftLogging.LogLevel.Trace => Level.Trace,
+        MicrosoftLogging.LogLevel.Debug => Level.Debug,
+        MicrosoftLogging.LogLevel.Information => Level.Info,
+        MicrosoftLogging.LogLevel.Warning => Level.Warn,
+        MicrosoftLogging.LogLevel.Error => Level.Error,
+        MicrosoftLogging.LogLevel.Critical => Level.Critical,
+        MicrosoftLogging.LogLevel.None => Level.Off,
+        _ => throw new ArgumentException(string.Format("LogLevel does not support value {0}.", logLevel), "logLevel")
+    };
   }
 
-  public bool IsEnabled (Microsoft.Extensions.Logging.LogLevel logLevel)
+  public static Level Convert (RemotionLogging.LogLevel logLevel)
   {
-    throw new NotImplementedException();
+    return logLevel switch
+    {
+        RemotionLogging.LogLevel.Debug => Level.Debug,
+        RemotionLogging.LogLevel.Info => Level.Info,
+        RemotionLogging.LogLevel.Warn => Level.Warn,
+        RemotionLogging.LogLevel.Error => Level.Error,
+        RemotionLogging.LogLevel.Fatal => Level.Fatal,
+        _ => throw new ArgumentException(string.Format("LogLevel does not support value {0}.", logLevel), "logLevel")
+    };
+  }
+
+  public void Log<TState> (MicrosoftLogging.LogLevel logLevel, MicrosoftLogging.EventId eventId, TState state, Exception? exception, Func<TState, Exception?, string> formatter)
+  {
+    var level = Convert(logLevel);
+    if (_logger.IsEnabledFor(level))
+      _logger.Log(CreateLoggingEvent(level, eventId.Id, formatter(state, exception), exception));
+  }
+
+  public bool IsEnabled (MicrosoftLogging.LogLevel logLevel)
+  {
+    return _logger.IsEnabledFor(Convert(logLevel));
   }
 
   public IDisposable? BeginScope<TState> (TState state)
       where TState : notnull
   {
-    throw new NotImplementedException();
+    return null;
+  }
+
+  private LoggingEvent CreateLoggingEvent (Level level, int? eventID, object? message, Exception? exceptionObject)
+  {
+    var loggingEvent = new LoggingEvent(typeof(Log4NetLogger), null, _logger.Name, level, message, exceptionObject);
+
+    if (eventID.HasValue)
+    {
+      if (eventID < 0 || eventID > 0xFFFF)
+      {
+        LogLoggingError(eventID.Value, exceptionObject, message);
+
+        throw new ArgumentOutOfRangeException(
+            "eventID", string.Format("An event id of value {0} is not supported. Valid event ids must be within a range of 0 and 65535.", eventID));
+      }
+
+      loggingEvent.Properties["EventID"] = eventID;
+    }
+
+    return loggingEvent;
+  }
+
+  private void LogLoggingError (int eventID, Exception? exceptionObject, object? message)
+  {
+    int safeEventID;
+    if (eventID < 0)
+      safeEventID  = 0;
+    else if (eventID > 0xFFFF)
+      safeEventID  = 0xFFFF;
+    else
+      safeEventID = eventID;
+
+    Level level = Level.Error;
+    if (_logger.IsEnabledFor(level))
+    {
+      if (_logger.IsEnabledFor(level))
+      {
```

Comment:
This is a copy of the original code but since we are loosing the trackable history, we should remove the redundancy to avoid questions about this in the future.